### ISSUE TITLE
Pytorch Complex Autograd used in cplxmodule?

### ISSUE THREAD

**Issue by Ujjawal-K-Panchal** [ivannz/cplxmodule#19]
I recently read about Pytorch autograd supporting complex differentiation using Wirtinger (CR) Calculus [here](https://pytorch.org/docs/master/notes/autograd.html#complex-autograd-doc) for it;s torch.complex datatype. However, the same document also talks about a different split way of computing gradients. I have been using cplxmodule, however, I'm not sure about calculation of gradients in the framework. Does the cplxmodule use the Pytorch complex autograd differentiation for calculation of gradients for a Cplx Tensors?

In summary, the complex autograd gradient is given by:
![torch-cplx-diff-1](https://user-images.githubusercontent.com/31011628/125159046-8e12be80-e192-11eb-9555-55111481fcdb.png)

where:
![torch-cplx-diff-2](https://user-images.githubusercontent.com/31011628/125159065-a71b6f80-e192-11eb-8044-e5e6eca4078a.png)

Hence, the entire eqn. reduces to the following eqn., since the `backward()` = 1 for scalar output.
![torch-cplx-diff-3](https://user-images.githubusercontent.com/31011628/125159070-ae427d80-e192-11eb-81a8-ae1a412149ae.png)

**Comment by ivannz** [comment]
No cplxmodule does not use torch's complex dtype. In issue #2 I outline [the mathematics](https://github.com/ivannz/cplxmodule/issues/2#issuecomment-622503453) behind the grads computed in `cplxmodule`. Briefly, we emulate the complex arithmetic and derivation, and compute the final `dL / d conj(z)` for a real-valued loss w.r.t. parameter $z$, that can be seamlessly used in standard optimiers in [`torch.optim.*`](https://pytorch.org/docs/master/notes/autograd.html#how-is-wirtinger-calculus-useful-in-optimization). Internally the computation is perfomed using way 1 from [here](https://pytorch.org/docs/master/notes/autograd.html#how-can-i-write-my-own-derivative-formula-for-a-complex-function) : Wirtinger calculus is convenient for computing the grads by hand, but the split approach allows leveraging existing real-valued autograd.

**Comment by Ujjawal-K-Panchal** [comment]
Thanks for the answer @ivannz!
